### ISSUE TITLE
How is missing value in labels handled when do regression?

### ISSUE THREAD

**Issue by Kexiii** [microsoft/LightGBM#6173]
Hi, I want to know how does lightgbm handle missing value(NAN) in target data when do regression? Is there any flag to drop the sample if the label is NAN?

**Comment by jameslamb** [comment]
Thanks for using LightGBM.

Missing values in the target are replaced with 0. The following example using the Python package demonstrates this:

```python
import lightgbm as lgb
import numpy as np

dtrain = lgb.Dataset(
    data=np.array([1.1, 1.2, 1.3], dtype=np.float32).reshape(3, 1),
    label=np.array([np.nan, 5.0, 6.0], dtype=np.float32),
    params={"min_data_in_bin": 1}
)
dtrain.construct()

dtrain.get_field("label")
# array([0., 5., 6.], dtype=float32)
```

That happens right here:

https://github.com/microsoft/LightGBM/blob/fcf76bceb902a69e1af7cdeac289d31916ddc3f5/src/io/metadata.cpp#L417-L419

https://github.com/microsoft/LightGBM/blob/fcf76bceb902a69e1af7cdeac289d31916ddc3f5/include/LightGBM/utils/common.h#L653-L663

> Is there any flag to drop the sample if the label is NAN?

That behavior is not configurable.

There is an open feature request to add a warning when this conversion happens, but as far as I know it isn't currently being worked on: #4483.

If you'd like to avoid those observations affecting the final model, consider one of the following approaches:

* remove them from the raw training data prior to passing that data to LightGBM
* supply `weights` when constructing the LightGBM `Dataset`, and set the weight of those samples to 0
    - *NOTE: feature values in those samples will still impact the learned bin boundaries in LightGBM's histogram, and those rows will still count toward count-based parameters like `min_data_in_leaf`*

**Comment by Kexiii** [comment]
Thanks, it's helpful
